// @passageidentity/passage-node/lib/esm/generated/models/Link.js
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: 1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the Link interface.
 */
export function instanceOfLink(value) {
    let isInstance = true;
    isInstance = isInstance && "href" in value;
    return isInstance;
}
export function LinkFromJSON(json) {
    return LinkFromJSONTyped(json, false);
}
export function LinkFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'href': json['href'],
    };
}
export function LinkToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'href': value.href,
    };
}
