// @passageidentity/passage-node/lib/esm/generated/models/ListPaginatedUsersResponse.js
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: 1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { ListPaginatedUsersItemFromJSON, ListPaginatedUsersItemToJSON, } from './ListPaginatedUsersItem';
import { PaginatedLinksFromJSON, PaginatedLinksToJSON, } from './PaginatedLinks';
/**
 * Check if a given object implements the ListPaginatedUsersResponse interface.
 */
export function instanceOfListPaginatedUsersResponse(value) {
    let isInstance = true;
    isInstance = isInstance && "_links" in value;
    isInstance = isInstance && "created_before" in value;
    isInstance = isInstance && "limit" in value;
    isInstance = isInstance && "page" in value;
    isInstance = isInstance && "total_users" in value;
    isInstance = isInstance && "users" in value;
    return isInstance;
}
export function ListPaginatedUsersResponseFromJSON(json) {
    return ListPaginatedUsersResponseFromJSONTyped(json, false);
}
export function ListPaginatedUsersResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        '_links': PaginatedLinksFromJSON(json['_links']),
        'created_before': json['created_before'],
        'limit': json['limit'],
        'page': json['page'],
        'total_users': json['total_users'],
        'users': (json['users'].map(ListPaginatedUsersItemFromJSON)),
    };
}
export function ListPaginatedUsersResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        '_links': PaginatedLinksToJSON(value._links),
        'created_before': value.created_before,
        'limit': value.limit,
        'page': value.page,
        'total_users': value.total_users,
        'users': (value.users.map(ListPaginatedUsersItemToJSON)),
    };
}
