// @passageidentity/passage-node/lib/esm/generated/models/Nonce.js
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: 1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the Nonce interface.
 */
export function instanceOfNonce(value) {
    let isInstance = true;
    isInstance = isInstance && "nonce" in value;
    return isInstance;
}
export function NonceFromJSON(json) {
    return NonceFromJSONTyped(json, false);
}
export function NonceFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'nonce': json['nonce'],
    };
}
export function NonceToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'nonce': value.nonce,
    };
}
