// @passageidentity/passage-node/lib/esm/generated/models/PaginatedLinks.js
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: 1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { LinkFromJSON, LinkToJSON, } from './Link';
/**
 * Check if a given object implements the PaginatedLinks interface.
 */
export function instanceOfPaginatedLinks(value) {
    let isInstance = true;
    isInstance = isInstance && "first" in value;
    isInstance = isInstance && "last" in value;
    isInstance = isInstance && "next" in value;
    isInstance = isInstance && "previous" in value;
    isInstance = isInstance && "self" in value;
    return isInstance;
}
export function PaginatedLinksFromJSON(json) {
    return PaginatedLinksFromJSONTyped(json, false);
}
export function PaginatedLinksFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'first': LinkFromJSON(json['first']),
        'last': LinkFromJSON(json['last']),
        'next': LinkFromJSON(json['next']),
        'previous': LinkFromJSON(json['previous']),
        'self': LinkFromJSON(json['self']),
    };
}
export function PaginatedLinksToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'first': LinkToJSON(value.first),
        'last': LinkToJSON(value.last),
        'next': LinkToJSON(value.next),
        'previous': LinkToJSON(value.previous),
        'self': LinkToJSON(value.self),
    };
}
