// @passageidentity/passage-node/lib/cjs/generated/models/Link.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: 1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LinkToJSON = exports.LinkFromJSONTyped = exports.LinkFromJSON = exports.instanceOfLink = void 0;
/**
 * Check if a given object implements the Link interface.
 */
function instanceOfLink(value) {
    let isInstance = true;
    isInstance = isInstance && "href" in value;
    return isInstance;
}
exports.instanceOfLink = instanceOfLink;
function LinkFromJSON(json) {
    return LinkFromJSONTyped(json, false);
}
exports.LinkFromJSON = LinkFromJSON;
function LinkFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'href': json['href'],
    };
}
exports.LinkFromJSONTyped = LinkFromJSONTyped;
function LinkToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'href': value.href,
    };
}
exports.LinkToJSON = LinkToJSON;
