// @passageidentity/passage-node/lib/cjs/generated/models/Model403Error.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: 1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Model403ErrorToJSON = exports.Model403ErrorFromJSONTyped = exports.Model403ErrorFromJSON = exports.instanceOfModel403Error = exports.Model403ErrorCodeEnum = void 0;
/**
 * @export
 */
exports.Model403ErrorCodeEnum = {
    CannotCreateOrganizationBillingPortalSession: 'cannot_create_organization_billing_portal_session',
    CannotCreateTransaction: 'cannot_create_transaction',
    CannotDeleteAdmin: 'cannot_delete_admin',
    CannotDeleteOrganizationMember: 'cannot_delete_organization_member',
    CannotSelfUpdateOrganizationMember: 'cannot_self_update_organization_member',
    OperationNotAllowed: 'operation_not_allowed'
};
/**
 * Check if a given object implements the Model403Error interface.
 */
function instanceOfModel403Error(value) {
    let isInstance = true;
    isInstance = isInstance && "code" in value;
    isInstance = isInstance && "error" in value;
    return isInstance;
}
exports.instanceOfModel403Error = instanceOfModel403Error;
function Model403ErrorFromJSON(json) {
    return Model403ErrorFromJSONTyped(json, false);
}
exports.Model403ErrorFromJSON = Model403ErrorFromJSON;
function Model403ErrorFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': json['code'],
        'error': json['error'],
    };
}
exports.Model403ErrorFromJSONTyped = Model403ErrorFromJSONTyped;
function Model403ErrorToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'error': value.error,
    };
}
exports.Model403ErrorToJSON = Model403ErrorToJSON;
