// @passageidentity/passage-node/lib/cjs/generated/models/Nonce.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: 1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NonceToJSON = exports.NonceFromJSONTyped = exports.NonceFromJSON = exports.instanceOfNonce = void 0;
/**
 * Check if a given object implements the Nonce interface.
 */
function instanceOfNonce(value) {
    let isInstance = true;
    isInstance = isInstance && "nonce" in value;
    return isInstance;
}
exports.instanceOfNonce = instanceOfNonce;
function NonceFromJSON(json) {
    return NonceFromJSONTyped(json, false);
}
exports.NonceFromJSON = NonceFromJSON;
function NonceFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'nonce': json['nonce'],
    };
}
exports.NonceFromJSONTyped = NonceFromJSONTyped;
function NonceToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'nonce': value.nonce,
    };
}
exports.NonceToJSON = NonceToJSON;
