// @passageidentity/passage-node/lib/cjs/generated/models/UserSocialConnections.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: 1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserSocialConnectionsToJSON = exports.UserSocialConnectionsFromJSONTyped = exports.UserSocialConnectionsFromJSON = exports.instanceOfUserSocialConnections = void 0;
const runtime_1 = require("../runtime");
const AppleUserSocialConnection_1 = require("./AppleUserSocialConnection");
const GithubUserSocialConnection_1 = require("./GithubUserSocialConnection");
const GoogleUserSocialConnection_1 = require("./GoogleUserSocialConnection");
/**
 * Check if a given object implements the UserSocialConnections interface.
 */
function instanceOfUserSocialConnections(value) {
    let isInstance = true;
    return isInstance;
}
exports.instanceOfUserSocialConnections = instanceOfUserSocialConnections;
function UserSocialConnectionsFromJSON(json) {
    return UserSocialConnectionsFromJSONTyped(json, false);
}
exports.UserSocialConnectionsFromJSON = UserSocialConnectionsFromJSON;
function UserSocialConnectionsFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'apple': !(0, runtime_1.exists)(json, 'apple') ? undefined : (0, AppleUserSocialConnection_1.AppleUserSocialConnectionFromJSON)(json['apple']),
        'github': !(0, runtime_1.exists)(json, 'github') ? undefined : (0, GithubUserSocialConnection_1.GithubUserSocialConnectionFromJSON)(json['github']),
        'google': !(0, runtime_1.exists)(json, 'google') ? undefined : (0, GoogleUserSocialConnection_1.GoogleUserSocialConnectionFromJSON)(json['google']),
    };
}
exports.UserSocialConnectionsFromJSONTyped = UserSocialConnectionsFromJSONTyped;
function UserSocialConnectionsToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'apple': (0, AppleUserSocialConnection_1.AppleUserSocialConnectionToJSON)(value.apple),
        'github': (0, GithubUserSocialConnection_1.GithubUserSocialConnectionToJSON)(value.github),
        'google': (0, GoogleUserSocialConnection_1.GoogleUserSocialConnectionToJSON)(value.google),
    };
}
exports.UserSocialConnectionsToJSON = UserSocialConnectionsToJSON;
