// @passageidentity/passage-node/lib/cjs/generated/models/UserRecentEvent.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: 1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserRecentEventToJSON = exports.UserRecentEventFromJSONTyped = exports.UserRecentEventFromJSON = exports.instanceOfUserRecentEvent = void 0;
const UserEventStatus_1 = require("./UserEventStatus");
/**
 * Check if a given object implements the UserRecentEvent interface.
 */
function instanceOfUserRecentEvent(value) {
    let isInstance = true;
    isInstance = isInstance && "created_at" in value;
    isInstance = isInstance && "completed_at" in value;
    isInstance = isInstance && "id" in value;
    isInstance = isInstance && "ip_addr" in value;
    isInstance = isInstance && "status" in value;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "user_agent" in value;
    return isInstance;
}
exports.instanceOfUserRecentEvent = instanceOfUserRecentEvent;
function UserRecentEventFromJSON(json) {
    return UserRecentEventFromJSONTyped(json, false);
}
exports.UserRecentEventFromJSON = UserRecentEventFromJSON;
function UserRecentEventFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'created_at': (new Date(json['created_at'])),
        'completed_at': (json['completed_at'] === null ? null : new Date(json['completed_at'])),
        'id': json['id'],
        'ip_addr': json['ip_addr'],
        'status': (0, UserEventStatus_1.UserEventStatusFromJSON)(json['status']),
        'type': json['type'],
        'user_agent': json['user_agent'],
    };
}
exports.UserRecentEventFromJSONTyped = UserRecentEventFromJSONTyped;
function UserRecentEventToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'created_at': (value.created_at.toISOString()),
        'completed_at': (value.completed_at === null ? null : value.completed_at.toISOString()),
        'id': value.id,
        'ip_addr': value.ip_addr,
        'status': (0, UserEventStatus_1.UserEventStatusToJSON)(value.status),
        'type': value.type,
        'user_agent': value.user_agent,
    };
}
exports.UserRecentEventToJSON = UserRecentEventToJSON;
