// @passageidentity/passage-node/lib/cjs/generated/models/UserEventStatus.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Passage Management API
 * Passage\'s management API to manage your Passage apps and users.
 *
 * The version of the OpenAPI document: 1
 * Contact: support@passage.id
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserEventStatusToJSON = exports.UserEventStatusFromJSONTyped = exports.UserEventStatusFromJSON = exports.UserEventStatus = void 0;
/**
 *
 * @export
 */
exports.UserEventStatus = {
    Complete: 'complete',
    Incomplete: 'incomplete'
};
function UserEventStatusFromJSON(json) {
    return UserEventStatusFromJSONTyped(json, false);
}
exports.UserEventStatusFromJSON = UserEventStatusFromJSON;
function UserEventStatusFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.UserEventStatusFromJSONTyped = UserEventStatusFromJSONTyped;
function UserEventStatusToJSON(value) {
    return value;
}
exports.UserEventStatusToJSON = UserEventStatusToJSON;
